contentcuration/contentcuration/management/commands/mark_incomplete.py

```diff
@@ -11,62 +11,64 @@
-        i = 0
-        count = 0
-        node_ids = query[i:i + CHUNKSIZE]
-        while node_ids:
-            ContentNode.objects.filter(pk__in=node_ids).update(complete=complete)
-            i += CHUNKSIZE
-            count += len(node_ids)
-            node_ids = query[i:i + CHUNKSIZE]
-        return count
+    def add_arguments(self, parser):
+        parser.add_argument("--chunksize", dest="chunksize", default=CHUNKSIZE)
 
     def handle(self, *args, **options):
+        chunksize = options["chunksize"]
+
         start = time.time()
 
-        # Mark invalid topics
-        topicstart = time.time()
-        logging.info('Marking topics...')
-        query = ContentNode.objects.filter(kind_id=content_kinds.TOPIC, title='', complete__isnull=True).values_list('id', flat=True)
-        count = self.mark_complete_field(query)
-        logging.info('Marked {} invalid topics (finished in {})'.format(count, time.time() - topicstart))
+        # Mark invalid titles
+        titlestart = time.time()
+        logging.info('Marking blank titles...')
+        count = ContentNode.objects.filter(title='').order_by().update(complete=False)
+        logging.info('Marked {} invalid titles (finished in {})'.format(count, time.time() - titlestart))
+
+        # Mark invalid licenses
+        licensestart = time.time()
+        logging.info('Marking blank licenses...')
+        count = ContentNode.objects.exclude(kind_id=content_kinds.TOPIC).filter(license__isnull=True).order_by().update(complete=False)
+        logging.info('Marked {} invalid licenses (finished in {})'.format(count, time.time() - licensestart))
+
+        licensestart = time.time()
+        logging.info('Marking blank license descriptions...')
+        custom_licenses = list(License.objects.filter(is_custom=True).values_list("pk", flat=True))
+        count = ContentNode.objects.exclude(kind_id=content_kinds.TOPIC)\
+            .filter(license_id__in=custom_licenses).filter(Q(license_description__isnull=True) | Q(license_description=''))\
+            .order_by().update(complete=False)
+        logging.info('Marked {} invalid license descriptions (finished in {})'.format(count, time.time() - licensestart))
 
         # Mark invalid file resources
         resourcestart = time.time()
         logging.info('Marking file resources...')
         i = 0
         count = 0
-        file_check_query = File.objects.filter(preset__supplementary=False, contentnode=OuterRef("id"))
+        file_check_query = File.objects.filter(preset__supplementary=False, contentnode=OuterRef("id")).order_by()
         query = ContentNode.objects \
             .exclude(kind_id=content_kinds.TOPIC) \
             .exclude(kind_id=content_kinds.EXERCISE) \
-            .filter(complete__isnull=True) \
-            .values_list('id', flat=True)
-        node_ids = query[i:i + CHUNKSIZE]
-        while node_ids:
+            .order_by()
+        total = query.count()
```

Comment:
Was there a performance concern around running count?

Reply:
Possibly - although it allows us to do only one query per loop, rather than two, and as each one was taking 20 seconds each, it seemed like even an upfront hit on the count query might be worth it.